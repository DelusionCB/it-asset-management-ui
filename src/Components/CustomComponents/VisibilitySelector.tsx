import * as React from 'react';
import {Col, FormGroup, Input, Label} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'
import {VisibilitySelectorProps} from './types.customcomps';

function VisibilitySelector ({label, validation, value, onChange, options, id, placeholder, disabled = false, invalid = false}: VisibilitySelectorProps): JSX.Element {
    const {t} = useTranslation()

    function getOptions (options: string[]): JSX.Element[] {
        return options.map((value, index) => {
            return (
                <option key={index} value={value}>{t(`visibility.${value}`)}</option>
            )
        })
    }

    return (
        <Col className='textfield'>
            <FormGroup>
                <Label for={id}>
                    {t(label)}
                </Label>
                <Input
                    id={id}
                    disabled={disabled}
                    invalid={invalid}
                    aria-invalid={invalid}
                    name={id}
                    placeholder={`${t(placeholder)}`}
                    type='select'
                    value={value}
                    onChange={(e) => { onChange(e); }}
                >
                    {getOptions(options)}
                </Input>
            </FormGroup>
        </Col>
    );
}

export default VisibilitySelector;
