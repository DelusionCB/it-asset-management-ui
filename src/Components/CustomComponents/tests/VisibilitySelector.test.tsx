import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import VisibilitySelector from '../VisibilitySelector';
import '@testing-library/jest-dom'

describe('<VisibilitySelector  />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <VisibilitySelector value='hidden' id={'id'} options={['one', 'two']} invalid={false} label='label' onChange={jest.fn()} placeholder={'placeholder'} disabled={false}/>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
