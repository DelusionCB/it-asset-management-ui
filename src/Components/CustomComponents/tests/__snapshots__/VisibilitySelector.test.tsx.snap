// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisibilitySelector  /> Snapshot renders correctly 1`] = `
<div
  className="textfield col"
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="id"
    >
      label
    </label>
    <select
      aria-invalid={false}
      className="form-select"
      disabled={false}
      id="id"
      name="id"
      onChange={[Function]}
      placeholder="placeholder"
      value="hidden"
    >
      <option
        value="one"
      >
        visibility.one
      </option>
      <option
        value="two"
      >
        visibility.two
      </option>
    </select>
  </div>
</div>
`;
